src/shared/pcre2-util.h

```diff
@@ -3,21 +3,20 @@
 
 #include "macro.h"
 
-#if HAVE_PCRE2
+typedef void pcre2_pattern;
```

Comment:
typedeffing this to void kinda sucks, since it disables all type-checking...

I think it would be nicer to define this as anonymous struct instead. This then means you have to cast, but that should be safe if you do it in a static/inline cast function.

Reply:
or just define it either against the anonymous struct or against the real type, depending on HAVE_PCRE2. After all, you do that in a header file specific to pcre2-util, where it shouldn't matter...